app/controllers/users_controller.rb

```diff
@@ -31,22 +31,23 @@ def new
 
   def create
     @user = User.new(user_params_with_ip)
+    if @user.valid?
+      if @user.spam? && @user.save_without_session_maintenance
```

Comment:
@sudara How about this? 
We COULD move it inside the actual user.save_with_session_maintenance block on line 44 but I like being able to check for spam right away.
Confirmed that this action won't save the user if it's not spam :)
```
2.6.0 :001 > false && true
 => false
```